src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -220,10 +216,16 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
         'node-output-data',
         (data) => {
             if (data) {
-                setOutputDataMap((prev) => new Map([...prev, [data.nodeId, data.data]]));
+                setOutputDataMap((prev) => {
+                    const existingData = prev.get(data.nodeId);
+                    if (!existingData || !isDeepEqual(existingData, data.data)) {
+                        return new Map([...prev, [data.nodeId, data.data]]);
+                    }
+                    return prev;
+                });
             }
         },
-        [setOutputDataMap]
+        [[...outputDataMap], setOutputDataMap]
```

Comment:
..... This is a deps array.

Reply:
Also, why do you still depend on `outputDataMap` at all?

Reply:
Try it yourself without doing this. It doesn't work.

Reply:
You are creating a new object with render. So this hook constantly updates.

Reply:
Oh

Reply:
Well something is wrong with this regardless

Reply:
You what? Fuck it. I can't even fix it using my usual invisible `div` hack.

Just make the deps array `[{}, setOutputDataMap]`. This should make it obvious that something is wrong.